Swiftification/Array.swift

```diff
@@ -29,4 +29,35 @@ public extension Array {
         return (indices ~= index ? self[index] : nil)
     }
     
+    /// Index of the first occurrence of item, if found.
+    func indexOf <U: Equatable> (item: U) -> Int? {
+        if item is Element {
+            return self.indexOf({ (object) -> Bool in
+                return (object as! U) == item
+            })
+        }
+        
+        return nil
+    }
+    
+    /// Returns the first occurence of item, if found
+    func find <U: Equatable> (item: U) -> Element? {
+        if let index: Int = indexOf(item) {
+            return self[index]
+        }
+        
+        return nil
+    }
+    
+    /// Randomly rearranges the elements of self using the Fisher-Yates shuffle
+    mutating func shuffle () {
```

Comment:
Instead of mutate, have this return the shuffled array. Or rename `shuffleInPlace` (patterned after Swift's sort methods). Also be sure to add tests (probably not to verify the result with shuffling ;), but run it a ton of times to make sure it doesn't crash).